src/panels/config/integrations/integration-panels/zwave_js/zwave_js-node-config.ts

```diff
@@ -0,0 +1,411 @@
+              .key=${id}
+              @change=${this._switchToggled}
+              .disabled=${!item.metadata.writeable}
+            ></ha-switch>
+          </div>
+        </div>
+      `;
+    }
+
+    if (item.configuration_value_type === "range") {
+      return html`${labelAndDescription}
+        <paper-input
+          type="number"
+          .value=${item.value}
+          .min=${item.metadata.min}
+          .max=${item.metadata.max}
+          .property=${item.property}
+          .propertyKey=${item.property_key}
+          .key=${id}
+          .disabled=${!item.metadata.writeable}
+          @value-changed=${this._numericInputChanged}
+        >
+        </paper-input> `;
+    }
+
+    if (item.configuration_value_type === "enumerated") {
+      return html`
+        ${labelAndDescription}
+        <div class="flex">
+          <paper-dropdown-menu
+            dynamic-align
+            .disabled=${!item.metadata.writeable}
+          >
+            <paper-listbox
+              slot="dropdown-content"
+              .selected=${item.value}
+              attr-for-selected="value"
+              .key=${id}
+              .property=${item.property}
+              .propertyKey=${item.property_key}
+              @iron-select=${this._dropdownSelected}
+            >
+              ${Object.entries(item.metadata.states).map(
+                ([key, state]) => html`
+                  <paper-item .value=${key}>${state}</paper-item>
+                `
+              )}
+            </paper-listbox>
+          </paper-dropdown-menu>
+        </div>
+      `;
+    }
+
+    return html`${labelAndDescription}
+      <p>${item.value}</p>`;
+  }
+
+  private _isEnumeratedBool(item) {
+    // Some Z-Wave config values use a states list with two options where index 0 = Disabled and 1 = Enabled
+    // We want those to be considered boolean and show a toggle switch
+    const disabledStates = ["disable", "disabled"];
+    const enabledStates = ["enable", "enabled"];
+
+    if (item.configuration_value_type !== "enumerated") {
+      return false;
+    }
+    if (!("states" in item.metadata)) {
+      return false;
+    }
+    if (!(0 in item.metadata.states) || !(1 in item.metadata.states)) {
+      return false;
+    }
+    if (
+      disabledStates.includes(item.metadata.states[0].toLowerCase()) &&
+      enabledStates.includes(item.metadata.states[1].toLowerCase())
+    ) {
+      return true;
+    }
+    return false;
+  }
+
+  private _switchToggled(ev) {
+    this._updateConfigParameter(ev.target, ev.target.checked ? 1 : 0);
+  }
+
+  private _dropdownSelected(ev) {
+    if (ev.target === undefined || this._config![ev.target.key] === undefined) {
+      return;
+    }
+    if (this._config![ev.target.key].value === ev.target.selected) {
+      return;
+    }
+
+    this._updateConfigParameter(ev.target, parseInt(ev.target.selected));
+  }
+
+  private debouncedUpdate = debounce((target) => {
```

Comment:
Z-Wave device updates actually send data over the Z-Wave network. I think that it might be better if we group the commands and require the user to press "Apply" ?

Reply:
My understanding is it takes a separate setValue call on the backend for each config item. If we batch them all and submit 5 or 6 at a time on a busy network it's possible one of the changes could get delayed or dropped